### ISSUE TITLE
Submitting binary data in request body to HTTP trigger causes error writing invocationRequest to worker ID...

### ISSUE THREAD

**Issue by ak37165** [Azure/azure-functions-dotnet-worker#533]
I have an HTTP triggered function where the request body contains binary data, such as an audio file.

The function code is never executed (no breakpoint is hit) and the console shows: 
```
 Error writing message type InvocationRequest to workerId: e2ebae92-186d-4ecf-8865-fb484e2253ac
[2021-07-10T02:20:19.657Z] System.Private.CoreLib: Error sending from server.
```

The issue happens on both localhost and when hosted in Azure. The issue does not happen for some smaller files, say up to 300kB.

The error is quite nasty because once it happens the function app instance stops responding to other HTTP requests. I have to restart the function app - so it affects not just the error-inducing request but all users of the function app after that.

**Comment by ak37165** [comment]
May have a bearing on https://github.com/Azure/azure-functions-dotnet-worker/issues/366 which deals with submitting files as form data.

**Comment by ak37165** [comment]
Not sure if this code is called by the function-worker at the host but seems request body may be attempted to be decoded using UTF8: https://github.com/Azure/azure-functions-host/blob/71a6ab35770c33165f1543c51397013d0446a32f/src/WebJobs.Script/Rpc/MessageExtensions/RpcMessageConversionExtensions.cs#L162

**Comment by brettsam** [comment]
Thank you for the report @ak37165 -- can you clarify -- if you use a 300kB binary data file, things work? So it seems to be due to size alone?

Can you share what version of the Worker package you are using in your project?

**Comment by ak37165** [comment]
Version information:

```
Azure Functions Core Tools
Core Tools Version:       3.0.3477 Commit hash: 5fbb9a76fc00e4168f2cc90d6ff0afe5373afc6d  (64-bit)
Function Runtime Version: 3.0.15584.0
```

Using 1MB (or 4MB as is more typical for my scenario) does cause the error to happen always. I tested with one 300kB file and that worked - but that was an image file instead of an audio file. So not sure if some binary characters could be the cause or just the size is the issue. Network speed should not matter as I tested both on localhost (client and function app on same computer) as well as hosted on Azure.

**Comment by ak37165** [comment]
Microsoft.Azure.Functions.Worker version 1.2.0

**Comment by satvu** [comment]
> Microsoft.Azure.Functions.Worker version 1.2.0

@ak37165 did you mean 1.1.0 or 1.3.0? ([Official releases](https://github.com/Azure/azure-functions-dotnet-worker/releases))

I tried to repro the bug, but haven't been able to do so successfully with the Azure Functions Core Tools version you mentioned and latest version of the worker.

**Comment by ak37165** [comment]
I did have both 1.1.0 in one project and 1.2.0 in another (even though its not listed in official releases - see screenshot in Test 2 below) .
Also attaching sample large file (PDF).

**TEST 1. Client is PostMan, Server (localhost) running without HTTPS (API was executed using the Visual Studio Debug button)**
Uploading files up to 2MB works, but sending a large file such as 4.3MB (sample file attached) causes the function to show "Executing" (in the Azure functions core tools console) but then never get to "Executed" (just hangs for a long time, several minutes at least). Will likely timeout in 10 minutes (set in host.json).

Microsoft.Azure.Functions.Worker: 1.1.0 
Microsoft.Azure.Functions.Worker.Sdk: 1.0.2

Code:
Below is the function code I am using (but this never hits a breakpoint set on the first line itself):

```
[Function("fileupload")]
        public async Task<HttpResponseData> FileUpload([HttpTrigger(AuthorizationLevel.Anonymous, "post",
            Route = "fileupload")] HttpRequestData req,
            FunctionContext executionContext)
        {
            var logger = executionContext.GetLogger("Function1");
            logger.LogInformation("C# HTTP trigger function processed a request.");            

            logger.LogInformation("file length: " + req.Body.Length);

            return req.CreateResponse(HttpStatusCode.OK);
        }
```
The client was PostMan, with POST request to the URL for the above function, Body set to "binary" and the file selected in the PostMan UI.

**Test 2. Server (localhost) with HTTPS, Client Javascript in an HTML page that creates the POST request to the API using XMLHttpRequest.**
Here, the API code is executed using the command (since Visual Studio Debug even with the command parameters set in the Project properties debug tab did not seem to support running with HTTPS):
`func host start -port 7071 --useHttps --pause-on-error --cert certificatefile.pfx --password PASWD`

The Azure Core tools shows error:
```
 Error writing message type InvocationRequest to workerId: 7b338ecd-3f39-487b-94be-07c7e6a1be32
[2021-09-11T14:15:46.817Z] System.Private.CoreLib: Error sending from server.
```
Here the Core tools version is 
Core Tools Version:       3.0.3477 Commit hash: 5fbb9a76fc00e4168f2cc90d6ff0afe5373afc6d  (64-bit)
Function Runtime Version: 3.0.15584.0

Microsoft.Azure.Functions.Worker version 1.2.0 (see screenshot).
![AzureWorker](https://user-images.githubusercontent.com/10171302/132951171-0191c4e0-1f75-48d0-9bb8-2c37dd060813.png)

Sample large file to try with:
[Pdf58Pages.pdf](https://github.com/Azure/azure-functions-dotnet-worker/files/7148005/Pdf58Pages.pdf)

**Comment by fabiocav** [comment]
@satvu is investigating this issue. Moving to spring 110 while the investigation is in progress.

**Comment by satvu** [comment]
Thanks for the detailed info! This bug was fixed in [1.3.0](https://github.com/Azure/azure-functions-dotnet-worker/releases/tag/1.3.0) ([see bug fix here](https://github.com/Azure/azure-functions-dotnet-worker/commit/704de76b5693bc24bdf6a9836ba8a109517c6a41)). 

Version 1.2.0 is mistakenly not tagged, so it did not appear on our repository's page of official releases. I believe [this is the commit that should have been tagged](https://github.com/Azure/azure-functions-dotnet-worker/commit/c96ebd2b67f47f539e21e4ef9ac5f17538b63d21).

To double check I started from the commit associated with 1.2.0 and tested out just the changes in the bug fix. After that, I was able to send large pdf files (4+ MB) and large audio files (WAV extension, 18MB) in the HTTP requests as binary data.

Closing this for now- if you update the worker to the latest version (or any version 1.3.0 and higher) and the issue persists please let us know.
